javaparser-core/src/main/java/com/github/javaparser/metamodel/InstanceOfExprMetaModel.java

```diff
@@ -30,5 +30,7 @@
 
     public PropertyMetaModel expressionPropertyMetaModel;
 
+    public PropertyMetaModel namePropertyMetaModel;
+
```

Comment:
Here is a good example that the metamodel also doesn't have the properties in the right order :-( But that's another issue.

Reply:
I'm not sure what you mean here? They seem to be sorted alphabetically, unless that's just coincidence.

Reply:
Oh, the whole getChildNodes thing. Those nodes should be listed in order of occurrence in source code (so expression goes before type goes before name.) I was thinking the metamodel could help, but that also doesn't have things in the right order. In my opinion the metamodel should be the source of all meta-information about a type of node, so it should have the correct order.

Right now the Concrete Syntax Model has the right order, but that thing should be mostly redundant when you have the metamodel already.

Reply:
I think the order generated here is based on one of the bigger metamodel classes (it's missing the generated tags, but the one where the constructor parameters get added to the metamodel etc, and there's fields are initialised). 

That is where the sorting is applied, and presumably propagates out leading to this side effect we see here with the fields being ordered (they're added here in the order that they exist within the list).

That's probably where I'd start.

Reply:
Yeah, it seems that sorting there wasn't such a good idea in retrospect. I probably made that, but I wasn't aware of these requirements yet :-)